client/src/app/containers/Game/useGame.ts

```diff
@@ -0,0 +1,235 @@
+            break;
+        case 'BuildCrib': {
+            const cribAction = currentAction as CribAction;
+            request.a = {
+                cs: cribAction.selectedCards
+                    ? cribAction.selectedCards.map(cardToGolangCard)
+                    : [],
+            };
+            break;
+        }
+        case 'Cut':
+            request.a = { p: (currentAction as CutAction).percCut };
+            break;
+        case 'Pegging': {
+            const pegAction = currentAction as PegAction;
+            if (
+                !pegAction.selectedCards ||
+                pegAction.selectedCards.length !== 1
+            ) {
+                request.a = { sg: true };
+            } else {
+                request.a = { c: cardToGolangCard(pegAction.selectedCards[0]) };
+            }
+            break;
+        }
+        case 'Counting':
+            request.a = { pts: (currentAction as CountHandAction).points };
+            break;
+        case 'CribCounting':
+            request.a = { pts: (currentAction as CountCribAction).points };
+            break;
+        default:
+            request.a = {};
+            break;
+    }
+    return request;
+}
+
+export function useGame(): Result {
+    const gameState = useSelector((state: RootState) => state.game);
+    const { currentUser } = useAuth();
+    const { setAlert } = useAlert();
+    const dispatch = useDispatch();
+
+    const fetchGame = async (id: number) => {
+        const response = await axios.get<Game>(
+            `/game/${id}?player=${currentUser.id}`,
+        );
+        return response.data;
+    };
+
+    const getGame = async (id: number) => {
+        try {
+            const game = await fetchGame(id);
+            dispatch(actions.setGame(game));
+        } catch (err) {
+            setAlert(err.response.data, 'error');
+        }
+    };
+
+    const refreshGame = async () => {
+        dispatch(actions.setLoading(true));
+        try {
+            const game = await fetchGame(gameState.currentGameID);
+            dispatch(actions.setGame(game));
+        } catch (err) {
+            setAlert(err.response.data, 'error');
+        }
+        dispatch(actions.setLoading(false));
+    };
+
+    const createGame = async (playerIDs: string[]) => {
+        dispatch(actions.setLoading(true));
+        try {
+            const createResult = await axios.post<CreateGameResponse>(
+                `/create/game`,
+                {
+                    playerIDs,
+                },
+            );
+            const getResult = await axios.get<Game>(
+                `/game/${createResult.data.id}`,
+            );
+            dispatch(actions.setGame(getResult.data));
+        } catch (err) {
+            setAlert(err.response.data, 'error');
+        }
+        dispatch(actions.setLoading(false));
+    };
+
+    const createActionHandler = (phase: Phase) => async (a: GameAction) => {
+        dispatch(actions.setLoading(true));
```

Comment:
:mag: did we `setLoading` around the action handler previously?

Reply:
This concept didn't exist previously; I'm doing it here because we're interacting with the network by `POST`ing to `/action` and also refreshing the game